tests/unit_tests/utils/test_path_provider.py

```diff
@@ -192,18 +193,152 @@ def test_start_document_path_provider_sets_data_session_directory_default_to_tmp
     start_doc_default_data_session_directory: RunStart,
 ):
     pp = StartDocumentPathProvider()
-    pp.update_run(name="start", start_doc=start_doc_default_data_session_directory)
+    pp.run_start(name="start", start_document=start_doc_default_data_session_directory)
     path = pp("det")
 
     assert path == PathInfo(
         directory_path=PosixPath("/tmp"), filename="det-p01-22", create_dir_depth=0
     )
 
 
-def test_start_document_path_provider_update_called_with_different_document_skips(
+@pytest.fixture
+def stop_doc_default_template() -> dict:
+    return {
+        "name": "stop",
+        "doc": {
+            "run_start": "27c48d2f-d8c6-4ac0-8146-fedf467ce11f",
+            "time": 1741264732.96875,
+            "uid": "401ad197-5456-4a7d-ba5b-9cf8ad38d914",
+            "exit_status": "success",
+            "reason": "",
+            "num_events": {"primary": 1},
+        },
+    }
```

Comment:
I think there is a mismatch here between the [`RunStop`](https://blueskyproject.io/event-model/main/explanations/data-model.html#run-stop-document) that bluesky sends from the RunEngine and the [`DataEvent`](https://github.com/DiamondLightSource/blueapi/blob/b8b2adf5a86ac7e6d35f476529893fdd45d6b2c2/src/blueapi/core/bluesky_types.py#L98) wrapping the `RunStop` that we send from server to client.

```suggestion
    return {
        "run_start": "27c48d2f-d8c6-4ac0-8146-fedf467ce11f",
        "time": 1741264732.96875,
        "uid": "401ad197-5456-4a7d-ba5b-9cf8ad38d914",
        "exit_status": "success",
        "reason": "",
        "num_events": {"primary": 1},
    }
```